### PR TITLE
chore(ci): add env vars to control macrobenchmark warm up

### PR DESCRIPTION
**Pull request by igoragoli**
[APMSP-1452 Investigate increase in instability in python/macrobenchmarks after wsgi to gunicorn change](https://datadoghq.atlassian.net/browse/APMSP-1452)

Overview:

- Add k6 environment variables to control warm up to Python macrobenchmarks. Please refer to https://github.com/DataDog/benchmarking-platform/pull/106 for a description of changes.

Motivation:

- `uwsgi` had issues that affected Python macrobenchmarks. We switched `uwsgi` for `gunicorn`. `gunicorn` brought problems of its own, namely latency spikes in the beginning of experiments. To avoid measuring initial spikes, a warm up before the actual experiments had to be added.
- Please see detailed notes here: https://docs.google.com/document/d/19dZru9rYzqWMyeHow5Krlw27UvGj_VaR_CmA-35HFaM/edit?tab=t.8acydyg34pk0

Testing:

- Compared results with those from two weeks ago (before `uwsgi` was substituted by `gunicorn` and before benchmarks were unstable). [See thorough comparison](https://docs.google.com/document/d/19dZru9rYzqWMyeHow5Krlw27UvGj_VaR_CmA-35HFaM/edit?tab=t.8acydyg34pk0#heading=h.p0lakz619cdw).
- Ran smoke benchmarks after changes were made both in dd-trace-py and in benchmarking-platform. See [dashboard](https://ddstaging.datadoghq.com/dashboard/x79-z8u-xxh?fromUser=true&refresh_mode=paused&tpl_var_bp_branch%5B0%5D=igoragoli%2Fpython-macrobenchmarks-add-warmup&tpl_var_bp_commit_sha%5B0%5D=b7e2419f&tpl_var_ci_pipeline_id%5B0%5D=47156559&from_ts=1729611900000&to_ts=1729612800000&live=false).

## Checklist
- [x] PR author has checked that all the criteria below are met
- [x] The PR description includes an overview of the change
- [x] The PR description articulates the motivation for the change
- [x] The change includes tests OR the PR description describes a testing strategy
- [x] The PR description notes risks associated with the change, if any
- [x] Newly-added code is easy to change
- [x] The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- [x] The change includes or references documentation updates if necessary
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: .gitlab/macrobenchmarks.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ variables:
   # If you have a problem with Gitlab cache, see Troubleshooting section in Benchmarking Platform docs
   image: $MACROBENCHMARKS_CI_IMAGE
   script: |
-    git clone --branch python/macrobenchmarks https://gitlab-ci-token:${CI_JOB_TOKEN}@gitlab.ddbuild.io/DataDog/benchmarking-platform platform && cd platform
+    git clone --branch igoragoli/python-macrobenchmarks-add-warmup https://gitlab-ci-token:${CI_JOB_TOKEN}@gitlab.ddbuild.io/DataDog/benchmarking-platform platform && cd platform
```
Comment (gnufede) [inline .gitlab/macrobenchmarks.yml:27]: Once the other PR is merged, the branch for bp should be reverted, no?

```suggestion
    git clone --branch python/macrobenchmarks https://gitlab-ci-token:${CI_JOB_TOKEN}@gitlab.ddbuild.io/DataDog/benchmarking-platform platform && cd platform
```

**Review by gnufede — APPROVED** [review]

File: .gitlab/macrobenchmarks.yml (hunk 1 above)
Comment (igoragoli) [reply .gitlab/macrobenchmarks.yml]: Yep, thanks!

**Review by ddyurchenko — APPROVED** [review]

**Pull request merged**